include/variant_detection/variant_detection.hpp

```diff
@@ -3,10 +3,15 @@
 #include <seqan3/std/filesystem>    // for filesystem
 #include <map>
 #include <vector>
+#include <fstream>
 
 #include "iGenVar.hpp"              // for struct cmd_arguments
 #include "structures/junction.hpp"  // for class Junction
 
+#include "bamit/all.hpp"
+#include "cereal/types/memory.hpp"
+#include "cereal/types/vector.hpp"
+
```

Comment:
Please move all includes, which you don't need in the hpp to the cpp file, such that they are only loaded, when the cpp is needed.